packages/web3-validator/src/validation.ts

```diff
@@ -0,0 +1,138 @@
+import { isHexStrict, hexToNumber, isAddress as isAddressUtil } from 'web3-utils';
+import { parseBaseType } from './utils';
+import { ValidInputTypes } from './types';
+
+export const isValidEthType = (type: string): boolean => {
+	const { baseType, baseTypeSize } = parseBaseType(type);
+
+	if (baseType === type) {
+		return true;
+	}
+
+	if ((baseType === 'int' || baseType === 'uint') && baseTypeSize) {
+		if (baseTypeSize <= 256 && baseTypeSize % 8 === 0) {
+			return true;
+		}
+
+		return false;
+	}
+
+	if (baseType === 'bytes' && baseTypeSize) {
+		if (baseTypeSize >= 1 && baseTypeSize <= 32) {
+			return true;
+		}
+
+		return false;
+	}
+
+	return false;
+};
+
+export const isBoolean = (value: ValidInputTypes) => {
+	if (!['number', 'string', 'boolean'].includes(typeof value)) {
+		return false;
+	}
+
+	if (typeof value === 'boolean') {
+		return true;
+	}
+
+	if (typeof value === 'string' && !isHexStrict(value)) {
+		return value === '1' || value === '0';
+	}
+
+	if (typeof value === 'string' && isHexStrict(value)) {
+		return value === '0x1' || value === '0x0';
+	}
+
+	if (typeof value === 'number') {
+		return value === 1;
+	}
+
+	return false;
+};
+
+export const isString = (value: ValidInputTypes) => typeof value === 'string';
+
+export const isUInt = (value: ValidInputTypes, type: string) => {
```

Comment:
```suggestion
export const isUint = (value: ValidInputTypes, type: string) => {
```

This change might cause undefined errors if `isUInt` is used elsewhere

Reply:
Is this file generated by `ajv`?

Reply:
No its our code file.